engine/src/main/java/org/hibernate/validator/internal/xml/stax/AbstractStaxBuilder.java

```diff
@@ -64,7 +64,13 @@ public boolean process(XMLEventReader xmlEventReader, XMLEvent xmlEvent) {
 	 */
 	protected String readSingleElement(XMLEventReader xmlEventReader) throws XMLStreamException {
 		// trimming the string value as it might contain leading/trailing spaces or \n
-		return xmlEventReader.nextEvent().asCharacters().getData().trim();
+		XMLEvent xmlEvent = xmlEventReader.nextEvent();
+		StringBuilder stringBuilder = new StringBuilder( xmlEvent.asCharacters().getData() );
+		while ( xmlEventReader.peek().isCharacters() ) {
+			xmlEvent = xmlEventReader.nextEvent();
+			stringBuilder.append( xmlEvent.asCharacters().getData() );
+		}
+		return stringBuilder.toString().trim();
```

Comment:
Could we maybe move this change to the first commit? It looks like it should have been there from the beginning.